### ISSUE TITLE
Reorganize scripts/

### ISSUE THREAD

**Issue by seantalts** [stan-dev/stanc3#797]
I think we can probably merge a few of the scripts and slim things down a bit. I think a fine flow might be:

1. Install opam - this isn't idempotent and will just reinstall over whatever you've got. It also has interactive prompts that need answering. This is different for windows vs. linux
2. Install 4.07.1 OCaml compiler switch (errors if already installed), pin build, js, and dev deps (idempotent). this is the same on Linux and Windows.
3. (optional) a script with Windows cross-compile dependencies
4. (optional) .bat script for Windows WSL builds (do we use this?)

This will essentially throw everything most developers care about into the 2nd script and let devs re-run that in case we've messed up our environment (:D) at the expense of making some of the Jenkins builds for those steps slightly longer. I think it will be pretty minimal since these will all be built in parallel and there aren't that many additional dev or JS dependencies. I think at the moment it's a bit tricky to figure out what you need to run to get up to speed.

Any thoughts on this? I'm happy to take a crack at it if people think this is a good idea. We can see how much longer Jenkins builds take on the PR. If it adds <1m I think I'm happy.

**Comment by rok-cesnovar** [comment]
Agree with the list and reorganization.

4 is not needed for our CI. We just use Linux instances with the cross compiler.

**Comment by rok-cesnovar** [comment]
We used that for our CI, where we built windows binaries with WSL for some reason :)

Me an Nic then just switched that to linux instances that we have more of and are cheaper.

**Comment by seantalts** [comment]
Ahh, okay. Should we delete it?

**Comment by rok-cesnovar** [comment]
I think its not needed and we also dont test if  it actually works so yeah, just delete it imo.
